tutorial/README.md

```diff
@@ -0,0 +1,413 @@
+
+When you start the Redshift service you will first need to create the TICKT database and load it. Follow the instructions [here](http://docs.aws.amazon.com/redshift/latest/dg/cm-dev-t-load-sample-data.html) to create/load the TICKT database.
+
+I used a Redshift database running on a 2 node cluster. Each node manages 2 slices for a total of 4 slices. This *usually* (read Redshift [documentation](http://docs.aws.amazon.com/redshift/latest/dg/c_high_level_system_architecture.html) for a more nuanced discussion) means that each table will be stored in 4 separate partitions, one for each slice. 
+
+## Usage##
+
+We are ready to interact with Redshift using the spark-redshift library. The skeleton of the program we will be using is shown in Listing 1. The entire `SparkRedshiftTutorial.scala` program can be accessed from [here](SparkRedshiftTutorial.scala). You can also use the Spark REPL to run the lines listed in the program below. 
+
+```scala
+package com.databricks.spark.redshift.tutorial
+import org.apache.spark.{SparkConf,SparkContext}
+import org.apache.spark.sql.SaveMode
+import java.io.File
+import org.apache.spark.sql.types.{StructType,StructField,StringType,IntegerType,LongType,DecimalType}
+
+
+object SparkRedshiftTutorial {
+  /*
+   * For Windows Users only
+   * 1. Download contents from link https://github.com/srccodes/hadoop-common-2.2.0-bin/archive/master.zip
+   * 2. Unzip the file in step 1 into your %HADOOP_HOME%/bin. In the example below  %HADOOP_HOME%=C:/MySoftware/hdphome
+   * 3. After step 2 %HADOOP_HOME/bin should contain winutils.exe
+   * 
+  */
+  if(System.getProperty("os.name").startsWith("Windows")){
+    val hdphome:File = new File("C:/MySoftware/hdphome");
+    System.setProperty("hadoop.home.dir",hdphome.getAbsolutePath().toString())    
+  }
+  
+  def main(args:Array[String]):Unit = {
+    val awsAccessKeyId = args(0) 
+    val awsSecretAccessKey = args(1)    
+    val redshiftDBName = args(2)
+    val redshiftUserId= args(3)
+    val redshiftPassword= args(4)
+    /**
+      * Sample URL-swredshift.czac2vcs84ci.us-east-1.redshift.amazonaws.com:5439
+      */
+    val redshifturl = args(5) 
+    val jdbcURL = s"jdbc:redshift://$redshifturl/$redshiftDBName?user=$redshiftUserId&password=$redshiftPassword"
+
+    val sc = new SparkContext(new SparkConf().setAppName("SparkSQL").setMaster("local"))
+
+	/*Configure SparkContext to communicate with AWS*/
+	val tempS3Dir = "s3n://redshift-spark/temp/"
+    sc.hadoopConfiguration.set("fs.s3n.awsAccessKeyId", awsAccessKeyId)
+    sc.hadoopConfiguration.set("fs.s3n.awsSecretAccessKey", awsSecretAccessKey)
+
+	/*Create the SQL Context*/    
+    val sqlContext = new org.apache.spark.sql.SQLContext(sc)
+  }
+} 
+```
+Listing 1 : SparkRedshiftExamples.scala skeleton
```

Comment:
I'm not sure whether `Listing XXX` style makes sense for a Markdown-formatted tutorial document like this, since it's kind of tricky to get the listing labels to appear clearly. Can we restructure this to avoid the need for this style of listing?